modules/persistent-storage-csi-vsphere-dynamic-provisioning.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-vpshere.adoc
+
+[id="csi-vsphere-dynamic-provisioning"]
+= vSphere CSI dynamic provisioning
+
+Dynamic volume provisioning allows storage volumes to be created on-demand, eliminating the need for cluster administrators to pre-provision storage.
+
+Dynamic provisioning of persistent storage depends on the capabilities of
+the CSI driver and underlying storage back end. The provider of the CSI
+driver should document how to create a storage class in {product-title} and
+the parameters available for configuration.
+
+.Procedure
+
+. Create a default storage class that ensures all PVCs that do not require
+any special storage class are provisioned by the installed CSI driver.
++
+[source,shell]
+----
+kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: <storage-class> <1>
+    annotations:
+      storageclass.kubernetes.io/is-default-class: "true"
+  provisioner: csi.vsphere.vmware.com <2>
+  parameters:
+    storagepolicyname: "vSAN Default Storage Policy"  #Optional Parameter
+----
```

Comment:
We already create a CSI storageclass for vsphere driver. Also there is already a default storageclass created in the cluster for in-tree driver. Most likely creating multiple default storageclasses will lead to weird errors and should not be documented.